detector.go

```diff
@@ -21,7 +21,7 @@ const (
 	CodeDetectFail = 100
 )
 
-var ErrFail = errors.New("detection failed")
+var ErrFail = errors.New("No buildpack groups passed detection.\nPlease check that you are running against the correct path.")
```

Comment:
This seems like it's intended for UI/UX. It doesn't follow the standard go conventions (ex. start with logger case). The fact that it has a new line character is a little strange too. 

I'm curious if a better approach would be to write out these messages as part of the `cmd/detector` by checking if `err == ErrorFail` somewhere around here: https://github.com/buildpack/lifecycle/blob/9d0ea0f8f5b934f8dec2fbb09e2f618e1c44c2b7/cmd/detector/main.go#L82-L84

Reply:
👍